sdl2-vga-terminal/src/TuiTerminal.cpp

```diff
@@ -0,0 +1,343 @@
+#include "TuiTerminal.hpp"
+#include <algorithm>
+#include "tui/utils.hpp"
+
+
+TuiTerminal::TuiTerminal(const std::string& title, const std::string& description):
+	_term(VgaTerminal("VgaTerminal TUI mode")), _title(title), _desc(description)
+{
+	_term.autoScroll = false;
+	_term.showCursor = false;
+}
+
+
+void TuiTerminal::draw()
+{
+	_term.resetViewport();
+	_drawBackGround();
+	_drawHeader();
+	_drawFooter();
+	_term.setViewPort(0, 5, _term.getMode().tw, _term.getMode().th - 5);
+}
+
+void TuiTerminal::render()
+{
+	_term.render();
+}
+
+void TuiTerminal::_drawBackGround() noexcept
+{
+	_term.fill(background_tc.c, background_tc.col, background_tc.bgCol);
+}
+
+void TuiTerminal::_drawHeader() noexcept
+{
+	// TODO aligment
+	// split in 3 parts, left, middle, right
+	// 3 rows, columns = 
+	constexpr uint8_t x = 4;
+	constexpr uint8_t y = 0;
+	constexpr uint8_t w = 72;
+	constexpr uint8_t h = 3;
+	constexpr uint8_t col = 14;
+	constexpr uint8_t bgCol = 1;
+	constexpr SDL_Rect r{ x, y, w, h };
+	
+	
+	const std::string version = std::string("Version ") + VERSION;
+	auto [lmark, rmark] = tui::align3Strings(_title, _desc, version, w);
+	const SDL_Rect rl = { x, y, lmark, h };
+	const SDL_Rect rm = { x+lmark-1, y, rmark-lmark+1, h };
+	const SDL_Rect rr = { x+rmark-1, y, w-rmark+1, h };
+
+	_drawLeftBorder(rl, col, bgCol, _title);
+	_drawMiddleBorder(rm, col, bgCol, _desc);
+	_drawRightBorder(rr, col, bgCol, version);
+	_drawShadow(r);
+}
+
+void TuiTerminal::drawSingleBorderDialog(const SDL_Rect& r, uint8_t col, uint8_t bgCol, const std::string& title, const std::string body)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'body' should be passed by const reference.](https://app.codacy.com/manual/Raffaello/sdl2-vga-terminal/pullRequest?prid=5472420)